### ISSUE TITLE
Headings to the left in Docs layout

### ISSUE THREAD

**Issue by dfahlander** [dexie/dexie-website-old#10]
I think we should use a layout with improved overview for dexie docs. It seems quite common to have a left navigation div with headers like *get started* *API reference* etc and put in the content to the right.

Simple layout
* http://i18next.com/docs/

React layout (more advanced, but still clean):
* https://facebook.github.io/react/docs/hello-world.html

I hope to fix this as soon as I have the time, unless anyone else feels eager to do it.

**Comment by nponiros** [comment]
Yes I also suggested something similar in https://github.com/dfahlander/dexie.js-web/pull/3 regarding the splitting of the docs into API, Tutorial etc. To do this in jekyll we will probably have to define categories for each page so that we can filter based on the category when creating the sidebar. Maybe there is also a way to get the full path and filter based on subdirectory. So if the path contains "API" we don't show the page in the sidebar but add it to the API page.

We will probably also need different layouts so that we can show all API docs in the API page sidebar, all Tutorial docs in a Tutorial page sidebar etc.
